src/main/java/net/tlabs/tablesaw/parquet/TablesawRecordConverter.java

```diff
@@ -385,15 +384,12 @@ public void addLong(long value) {
 
                             @Override
                             public void addBinary(final Binary value) {
-                                Preconditions.checkArgument(value.length() == BINARY_INSTANT_LENGTH_VALUE,
-                                    BINARY_INSTANT_LENGTH_MESSAGE);
+                                Preconditions.checkArgument(value.length() == BINARY_INSTANT_LENGTH_VALUE, BINARY_INSTANT_LENGTH_MESSAGE);
                                 final ByteBuffer buf = value.toByteBuffer();
                                 buf.order(ByteOrder.LITTLE_ENDIAN);
-                                final long nanoday = buf.getLong();
-                                final int julianday = buf.getInt();
-                                final LocalDate date = LocalDate.ofEpochDay(0).with(JulianFields.JULIAN_DAY, julianday);
+                                final LocalDate date = LocalDate.ofEpochDay(0).with(JulianFields.JULIAN_DAY, buf.getInt());
                                 proxy.appendInstant(colIndex, ZonedDateTime.of(date.atStartOfDay(), ZoneOffset.UTC)
-                                    .toInstant().plus(nanoday, ChronoUnit.NANOS));
+                                    .toInstant().plus(buf.getLong(), ChronoUnit.NANOS));
```

Comment:
That does not work, the long needs to be read from the buffer before the int. This is why it was written this way.

Reply:
Note that this piece of code is not tested due to the lack of test file using this format...

Reply:
>I think we should not make those classes final and/or private: this is

The open-ness of code has nothing to do with allowing a class to be extended and final should be the default if a class is not designed or easily extendable; delegation/composition should be used instead (it's a common mistake in nowadays OO design). 

This is an issue here because

- The tablesaw code is NOT designed to be extended in part because of the builders and the lack of proper interfaces.

- The ts-partquet code is inheriting from tablesaw classes, that is the origin of the DSL oddities I mentioned before. The tablesaw class should NOT be extended but used as delegates instead. 

- The functionality the readers/writers provide are not meant to be extended by sub-classing since it's not addressing a generic problem/situation but a specific one, and I don't see a situation where one would ever want to do that: if there is additions to be done to the readers/writers to support additional features, this should be done within this repository.

Reply:
> That does not work, the long needs to be read from the buffer before the int. This is why it was written this way.

Good catch.

Reply:
Easy catch: I made the exact same mistake previously  ;-)

Regarding class extension I generally agree with you, in particular regarding OO design. We will have to discuss this more globally (level of integration with tablesaw, extensibility, and so on). We need to take the decision before the 1.0 release: after that, high-level changes like this are more difficult to introduce. In the meantime, let's keep it that way and give people the opportunity to use this library and give us feedback.